src/bull/bull.service.ts

```diff
@@ -23,6 +23,12 @@ export class BullService {
     return this.queue.add('scrape-data', data);
   }
 
+  async sleepAtRandomInterval() {
+    const randomDelay = Math.floor(Math.random() * 5000) + 1000;
+    this.logger.log(`sleep for ${randomDelay / 1000} seconds`);
+    await new Promise((resolve) => setTimeout(resolve, randomDelay));
```

Comment:
Instead of using setTimeout, how about using one of the [Bull's queue options](https://github.com/OptimalBits/bull/blob/master/REFERENCE.md#queueadd):

<img width="1053" alt="image" src="https://github.com/nainglinaung/scraper-test/assets/696529/1a077c12-9335-41a5-9622-759c44934c22">

The delay would be defined at the enqueuing step process, thus removing that responsibility from the job process. It would be easier to test and debug 👨‍⚕️

Reply:
Good improvement in [08717ff](https://github.com/nainglinaung/scraper-test/pull/7/commits/08717ff5d311ee1f21dcdfcbb4a2ab73a955de08) 👍